src/libsystemd-network/icmp6-util.c

```diff
@@ -156,3 +157,74 @@ int icmp6_send_router_solicitation(int s, const struct ether_addr *ether_addr) {
 
         return 0;
 }
+
+int icmp6_receive(int fd, void *iov_base, size_t iov_len,
```

Comment:
this naming of "iov_base" and "iov_len" is a bit misleading. I understand where this comes form, but ultimately, from the caller's perspective these parameters have little to do with "iovecs", and hence should probably lose their "iov_" prefix. Maybe call them "buffer" and "size" or so?